user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -295,20 +297,22 @@ define([
       var config = myself.get("spinConfig").getLargeConfig();
       var spinner1 = new Spinner(config),
           spinner2 = new Spinner(config);
-      var _clickedFolder = undefined;
-      var _clickedFile = undefined;
-      if ( foldersTreeModel ) {
-        _clickedFolder = {
-          obj: foldersTreeModel.get("clickedFolder"),
-          time: (new Date()).getTime()
-        }
-      }
-      if ( fileListModel ) {
-        _clickedFile = {
-          obj: fileListModel.get("clickedFile"),
-          time: (new Date()).getTime()
-        }
-      }
+
+      var fileBrowserClickedFolder = myself.get("clickedFolder") &&
+      myself.get("clickedFolder").obj ? myself.get("clickedFolder").obj : null;
+      var fileBrowserClickedFile = myself.get("clickedFile") &&
+      myself.get("clickedFile").obj ? myself.get("clickedFile").obj : null;
+
+      var _clickedFolder = {
+        obj: foldersTreeModel ? foldersTreeModel.get("clickedFolder") : fileBrowserClickedFolder,
+        time: (new Date()).getTime()
+      };
+
+      var _clickedFile = {
+        obj: fileListModel ? fileListModel.get("clickedFile") : fileBrowserClickedFile,
+        time: (new Date()).getTime()
+      };
```

Comment:
Would be better to have something like:

```js
var _clickedFolderObj = (foldersTreeModel && foldersTreeModel.get("clickedFolder")) || 
         (myself.get("clickedFolder") && myself.get("clickedFolder").obj);

var _clickedFolder = _clickedFolderObj ? {obj: _clickedFolderObj, time: (new Date()).getTime()} : null;
```

Then, you shouldn't need to be testing for `clickedFolder && clickedFolder.obj` elsewhere.
I see you also test for `clickedFolder.obj.attr("path")` elsewhere. If that's really needed, then it seems that should also be included here.